Src/BootCamp.Chapter/Csv/CsvBase.cs

```diff
@@ -0,0 +1,117 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace BootCamp.Chapter.Csv
+{
+    public class CsvBase
+    {
+        public CsvRow Header { get; protected set; } = new CsvRow();
+        public string Footer { get; protected set; } = string.Empty;
+        public CsvDelimiter Delimiter { get; protected set; } = CsvDelimiter.Comma;
+        protected string FileName { get; set; }
+        protected bool HasHeader { get; set; }
+        protected bool HasFooter { get; set; }
+        protected IList<CsvRow> Rows { get; set; } = new List<CsvRow>();
+
+        public CsvBase(string fileName)
+        {
+            if (!fileName.IsValid())
+            {
+                throw new ArgumentException("fileName cannot be null or empty");
+            }
+
+            FileName = fileName;
+        }
+
+        public CsvBase(string fileName, CsvDelimiter delimiter, bool hasHeader)
+        {
+            if (!fileName.IsValid())
+            {
+                throw new ArgumentException("fileName cannot be null or empty");
+            }
+
+            FileName = fileName;
+            Delimiter = delimiter;
+            HasHeader = hasHeader;
+        }
+
+        public CsvBase(string fileName, CsvDelimiter delimiter, bool hasHeader, bool hasFooter)
+        {
+            if (!fileName.IsValid())
+            {
+                throw new ArgumentException("fileName cannot be null or empty");
+            }
+
+            FileName = fileName;
+            Delimiter = delimiter;
+            HasHeader = hasHeader;
+            HasFooter = hasFooter;
+        }
+
+        protected bool TryParseRow(string input, out CsvRow csvRow)
+        {
+            csvRow = new CsvRow();
+
+            if (!input.IsValid())
+            {
+                return false;
+            }
+
+            var builder = new StringBuilder();
+            var isQuote = false;
+
+            foreach (var letter in input)
```

Comment:
Split into a function of its' own. ``CreateRowString`` or similar.

Reply:
I thought about how to refactor it since I first wrote it couple of days ago but I don't see a proper way to do it. Can you shed some light on how to do it?

Reply:
```cs
protected bool TryParseRow(string input, out CsvRow csvRow)
{
	        if (string.IsNullOrWhitespace(input))
            {
                return false;
            }
			
			// have string builder inside Csvrow()
            row = new CsvRow();

            foreach (var letter in input)
            {
				
				// move this logic to row itself.
				// like this: row.TakeLetter(letter);
				var isQuote = letter == '\"';
                if(!isQuote)
				{
					// internally will use SB
					row.AppendLetter(letter);
				}
                else if (letter == (char)Delimiter)
                {
					// internally will use SB
                    row.NextWord();
                }
            }
			
			csvRow.Add(builder.ToString().Trim());
            return true;
        }
}
```